prov/gni/src/gnix_vector.c

```diff
@@ -107,47 +122,31 @@ static inline int __gnix_vec_create(gnix_vector_t *vec, gnix_vec_attr_t *attr)
 	return FI_SUCCESS;
 }
 
-static inline void __gnix_vec_close_entries(gnix_vector_t *vec)
-{
-	memset(vec->vector, 0, (sizeof(gnix_vec_entry_t) * vec->attr.cur_size));
-}
-
-static inline void __gnix_vec_close(gnix_vector_t *vec)
+static inline int __gnix_vec_close(gnix_vector_t *vec)
 {
 	free(vec->vector);
 	vec->ops = NULL;
 	vec->attr.cur_size = 0;
 	vec->state = GNIX_VEC_STATE_DEAD;
-}
-
-/*******************************************************************************
- * INLINE OPS FNS
- ******************************************************************************/
-inline int _gnix_vec_insert_first(gnix_vector_t *vec, gnix_vec_entry_t *entry)
-{
-	return _gnix_vec_insert_at(vec, entry, 0);
-}
 
-inline int _gnix_vec_insert_last(gnix_vector_t *vec, gnix_vec_entry_t *entry)
-{
-	return _gnix_vec_insert_at(vec, entry, vec->attr.cur_size-1);
+	return FI_SUCCESS;
```

Comment:
> You should probably be checking to ensure that the vector isn't already dead. This would prevent a double free crash.

This is done in `__gnix_vec_at`.